src/icon-manifest/supportedExtensions.ts

```diff
@@ -158,6 +158,7 @@ export const extensions: IFileCollection = {
     { icon: 'groovy2', extensions: [], languages: [languages.groovy], format: FileFormat.svg, disabled: true },
     { icon: 'grunt', extensions: ['gruntfile.js'], filename: true, format: FileFormat.svg },
     { icon: 'gulp', extensions: ['gulpfile.js'], filename: true, format: FileFormat.svg },
+    { icon: 'gulp', extensions: ['gulpfile.babel.js'], filename: true, format: FileFormat.svg },
```

Comment:
do not redefine the extension, just add it the line above like this:
`{ icon: 'gulp', extensions: ['gulpfile.js', 'gulpfile.babel.js'], filename: true, format: FileFormat.svg },`